wallet/invoices.c

```diff
@@ -400,6 +408,7 @@ void invoices_resolve(struct invoices *invoices,
 	invoice->pay_index = pay_index;
 	invoice->msatoshi_received = msatoshi_received;
 	invoice->paid_timestamp = paid_timestamp;
+	invoice->expiration_timer = tal_free(invoice->expiration_timer);
```

Comment:
You don't need to free this, as the == UNPAID check above works fine.  Maybe it's slightly more efficient to do this though, so I don't completely disagree.  In which case turn the UNPAID check in trigger_expiration into an assert()?

Reply:
Probably better as an assert, yes.